src/Atmos/Model/nonlinear.jl

```diff
@@ -0,0 +1,82 @@
+export NonLinear
+
+struct NonLinear <: ModelPart end
+
+# FIXME: this doesn't seem to work ?
+const use_linearized_pressure = false
+
+@inline function flux_advective!(m::AtmosModel{NonLinear},
+                                 flux::Grad, state::Vars, aux::Vars, t::Real)
+  ρ = state.ρ
+  ρu = state.ρu
+  ρe = state.ρe
+  
+  ρ_ref = aux.ref_state.ρ
+  ρu_ref = aux.ref_state.ρu
+  ρe_ref = aux.ref_state.ρe
+  
+  ρ_inv = 1 / ρ
+  u = ρ_inv * ρu
+  e = ρ_inv * ρe
+
+  ρ_ref_inv = 1 / ρ_ref
+  u_ref = ρ_ref_inv * ρu_ref
+  e_ref = ρ_ref_inv * ρe_ref
+
+  flux.ρ   = -zero(eltype(state))
+  flux.ρu  = ρu * (u' - u_ref')
```

Comment:
```suggestion
  flux.ρu  = ρu * (u - u_ref)'
```

Not sure why, but this was needed for me to run on the GPU